### ISSUE TITLE
General Improvements

### ISSUE THREAD

**Issue by vinanrra** [surajcm/darkhold#3]
Is there a way to only show the user that just join the challenge instead of all users that played/register?

![imagen](https://user-images.githubusercontent.com/4247187/117009144-b6da8b80-aceb-11eb-9994-8e7f2fe59ffd.png)

And when i choose a challenges doesnt show the correct questions and its showing me always the same question:

![imagen](https://user-images.githubusercontent.com/4247187/117009595-38321e00-acec-11eb-8a3e-63f3316f1759.png)

And i have choose my own question but isnt working :/

**Comment by surajcm** [comment]
* Currently there is no way to show only the current user because the idea was to mimic the Kahoot feel. But, I have plans to add a game configuration screen, from which the admin can set these values. I will eventually add it. 
* I haven't tested with a different set of questions. I will try to test it and get this issue fixed 👍

**Comment by vinanrra** [comment]
Maybe I haven't explained good enough, is there a way to only show the active users on the challenge instead of a global scoreboard, because its showing all users even the users not playing that challenge.

**Comment by surajcm** [comment]
Ohh... Now I got it.  I have noticed that, yes, that is something that needs to be fixed. I will try to resolve it.

Thank you for the screenshots and detailed explanations. This really helps 👍

**Comment by vinanrra** [comment]
Thanks for everything 💓

**Comment by vinanrra** [comment]
Thanks, I always send photos because that way it's easier to check the problem, and hope everything can be fixed, because it's a really nice tool :)

**Comment by surajcm** [comment]
I love to improve the product, so any feedback is welcome 😃 . I will try to fix it one by one 👍

**Comment by vinanrra** [comment]
I have tested the new additions and its working flawless, but today i didn't have the same question was working fine, maybe i can reproduce the error again because i have cleared all other challenges and maybe happens when you have multiple challenges.

**Comment by surajcm** [comment]
Awesome 🤩 . I haven't tried Ansible yet. But if it is a configuration inside the repository, please feel free to raise a PR.

**Comment by vinanrra** [comment]
I have created a Docker Image instead of an Ansible Playbook

**Comment by vinanrra** [comment]
One more thing that could be good it's a way to change the default admin account and a way to block new registrations or at least to block user to only see their own challenges.

**Comment by surajcm** [comment]
Sure.
1. I think your question is more about adding more admin users, currently, there is no user role specified while signing up, mainly because all the users including admin have the same level of access. But yes, there are plans to add more pages that can fine-tune the behavior of the game, and those pre-set values should only be accessed by admin users. I will add a screen inside the admin login to add new users (both admin and participants), also manage/ delete them.
2. I don't really get the use case for blocking the new registration. Maybe once the user management page on admin is added, it may be fine to remove the new registrations. 
Yes, ideally the user should see their own challenges only. I will address it. 👍

**Comment by vinanrra** [comment]
Oh, nice and yeah, if you add that isnt need it to block new registration.

**Comment by vinanrra** [comment]
I just remember that there is one more thing missing but in "gameplay" give points depending on "speed", being the first and being the faster (time), so you will get more points if you respond the first and in less time, instead of a fixed points.

**Comment by surajcm** [comment]
Yes, you are right. That's one of the core features of Kahoot. I always wanted to have that, My focus shifted to the completion of the app instead of the functionalities in between. I think it may not be a big work, I will take a look at it. Thanks a lot for pointing out that 👍

**Comment by surajcm** [comment]
by the way, I added some code changes that restrict the view of challenges created by one user to another. It is kind of non-backward compatible, it works fine in my local, not sure whether it will break in the docker environment. I will update the wiki with instructions on how to fix these non-backward compatible issues

**Comment by vinanrra** [comment]
> by the way, I added some code changes that restrict the view of challenges created by one user to another. It is kind of non-backward compatible, it works fine in my local, not sure whether it will break in the docker environment. I will update the wiki with instructions on how to fix these non-backward compatible issues

I will check it later, but shouldn't  break anything at the Docker.

**Comment by surajcm** [comment]
> I just remember that there is one more thing missing but in "gameplay" give points depending on "speed", being the first and being the faster (time), so you will get more points if you respond the first and in less time, instead of a fixed points.

I have added this feature now. Please check that and let me know the feedback. The points are not rounded (max point per question is 1000), to make it feel so real.

**Comment by vinanrra** [comment]
List of things, to don't forget 👅 

- [X] Per user challenges
- [X] Improve point system
- [X] Persistent database
- [ ] Admin panel (Manage users, challenges...)

I will test it later after work.

**Comment by vinanrra** [comment]
I have added the Wiki pages, I hope you like it

**Comment by surajcm** [comment]
I was not well for the last few days, now only I am checking my emails.

I checked the wiki pages it looks awesome 😍 . 

Yes, you are right, in Kahoot, the host doest participate in the challenge. And, yes, it would be good to stop the timer if everyone is done, also it is better to have a control over the timer. I will definitely take a look at it.

**Comment by vinanrra** [comment]
> 
> 
> I was not well for the last few days, now only I am checking my emails.
> 
> I checked the wiki pages it looks awesome 😍 .
> 
> Yes, you are right, in Kahoot, the host doest participate in the challenge. And, yes, it would be good to stop the timer if everyone is done, also it is better to have a control over the timer. I will definitely take a look at it.

Hope everything it's fine 😄

**Comment by surajcm** [comment]
I was down with COVD for the last few days. I am still not yet fully recovered. 

I will definitely take a look at the issues, once I am back.
